tests/lib/rules/no-useless-character-class.ts

```diff
@@ -47,6 +47,11 @@ tester.run("no-useless-character-class", rule as any, {
         String.raw`/\c[A]/`,
         String.raw`/\c[Z]/`,
         String.raw`/\c[m]/`,
+        String.raw`/[\q{abc}]/v`,
+        // This rule does not support them.
+        String.raw`/[^[abc]]/v`, // -> /[^abc]/v
+        String.raw`/[^[abc]d]/v`, // -> /[^abcd]/v
+        String.raw`/[[abc][def]ghi]/v`, // -> /[abcdefghi]/v
```

Comment:
I'm wondering if we should check these with this rule.

https://github.com/ota-meshi/eslint-plugin-regexp/issues/574#issuecomment-1712739799